pkg/commands/compute/deploy_test.go

```diff
@@ -16,25 +16,59 @@ import (
 )
 
 func TestDeploy(t *testing.T) {
+	// NOTE: Some tests don't provide a Service ID via any mechanism (e.g. flag
```

Comment:
The changes to this function can be summarised as follows...

1. Define a single temp environment rather than having a new environment for each test scenario.
2. Replace short flag `-s` with the long flag `--service-id`.
3. Use flag for setting Service ID and not the manifest (makes bullet point 1 easier).
4. Recreate manifest for each scenario, as the mutation of the manifest affects the test expectations.